### PR TITLE
chore: instrument smithy-client v2

### LINKED ISSUE #3523: @aws-sdk/client-s3 TAV tests failing in main
Latest TAV tests in main are failing for `@aws-sdk/client-s3` instrumentation see https://github.com/elastic/apm-agent-nodejs/actions/runs/5713633220/job/15479387000#step:3:288

We should check if there have been changes in higher versions that we are not aware of. The assertion failing is about the error reported to the APM server.

We expect to send a proper span with `failure` outcome.
```
{
  name: "S3 GetObject elasticapmtest-bucket-3",
  type: "storage",
  subtype: "s3",
  action: "GetObject",
  context:
    {
      service: { target: { type: "s3", name: "elasticapmtest-bucket-3" } },
      destination:
        {
          address: "localstack",
          port: 4566,
          cloud: { region: "us-east-2" },
          service: { type: "", name: "", resource: "elasticapmtest-bucket-3" },
        },
      http: { status_code: 404 },
    },
  otel:
    {
      attributes:
        {
          "aws.s3.bucket": "elasticapmtest-bucket-3",
          "aws.s3.key": "aDir/aFile.txt-does-not-exist",
        },
    },
  outcome: "failure"
}
```

But instead we get a HTTP span
```
{
  name: "GET localstack:4566",
  type: "external",
  subtype: "http",
  action: "GET",
  context:
    {
      service: { target: { type: "http", name: "localstack:4566" } },
      destination:
        {
          address: "localstack",
          port: 4566,
          service: { type: "", name: "", resource: "localstack:4566" },
        },
      http:
        {
          method: "GET",
          status_code: 404,
          url: "http://localstack:4566/elasticapmtest-bucket-3/aDir/aFile.txt-does-not-exist?x-id=GetObject",
        },
    },
  outcome: "failure"
}
```

### PR DESCRIPTION
**Pull request by david-luna**
Add support for `@smithy/smithy-client` v2+. This version is introduced by `@aws-sdk/client-s3` and others in version [v3.378.0](https://github.com/aws/aws-sdk-js-v3/commit/c705b976bb82852ead46da81eb0f2101266e9acb)

Looking at the major release of `smithy-client` there are breaking changes in its types but these are not affecting the middlewares of the AWS clients so its safe to add support for it.

Closes #3523

### Checklist

- [x] Implement code
- [ ] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Review by david-luna — COMMENTED** [review]
/test tav @aws-sdk/client-s3

**Review by david-luna — COMMENTED** [review]
/test tav @aws-sdk/client-s3 16

File: package-lock.json
Comment (trentm) [inline package-lock.json:1]: This accidentally changed to lockfileVersion: 1:

```diff
diff --git a/package-lock.json b/package-lock.json
index 51f534bf..bc50cde7 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -1,17985 +1,8 @@
 {
   "name": "elastic-apm-node",
   "version": "3.48.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
...
```

I think you may have `npm install`ed with an older version of Node than v16.

**Review by trentm — CHANGES_REQUESTED** [review]
Need to fix the package-lock file.

**Comment by trentm** [comment]
The huge change to `package-lock.json` here is partly due to the git diff algorithm. It is more sane with the "patience" algorithm:

```
% git diff main...  | wc -l
   35913

% git diff --patience main...  | wc -l
    7261
```

A lot of the diff is because these devDeps are at different versions:

```
    "@aws-sdk/client-s3": "^3.379.1",
    "@aws-sdk/s3-request-presigner": "^3.363.0",
```

Leading to a loss of a lot of possible dedupes for common transitive deps -- now at different versions. I'll upgrade the latter and see if that helps.

**Comment by trentm** [comment]
Yes, that helps a lot. For example, instead of >30 installations of `@smithy/types`, there is now just one:

```
% npm ls @smithy/types | rg -v deduped | rg smithy/types
│ ├── @smithy/types@2.0.2
```

And the diff is reduced:

```
% git diff --patience main...  | wc -l
    3648
```

and with the default algorithm as well:

```
% git diff  main...  | wc -l
    3660
```

@david-luna  **I think as a rule, if we update one `@aws-sdk/...` dep, we should update them all.**

**Review by trentm — APPROVED** [review]
LGTM, assuming tests pass again.

**Review by trentm — COMMENTED** [review]
/test tav @aws-sdk/client-s3

**Pull request merged**